### DISCUSSION TITLE
Why is `pad_to_sequence_len: true recommended when using sample_packing?

### DISCUSSION THREAD

**Discussion by RicardoDominguez** [axolotl-ai-cloud/axolotl#1101] [Q&A] [answered]
As far as I can see, when using sample_packing the effective batch size is 1, and the inputs are padded to a multiple of 64. Beyond this padding, why is it recommended to pad to the sequence len?

**Comment by winglian** [comment, marked as answer]
This is due to the fact that feeding varying batch sizes to the gpu a ten leeds to gpu memory actually increasing from having varying lengths even though they are still smaller than a fully padded batch.

**Comment by RicardoDominguez** [comment]
Thanks!
